lib/Auth/Process/privacyidea.php

```diff
@@ -104,67 +104,77 @@ public function process(&$state)
-	    curl_setopt($curl_instance, CURLOPT_RETURNTRANSFER, TRUE);
-	    curl_setopt($curl_instance, CURLOPT_USERAGENT, "simpleSAMLphp");
-	    // Add POST params
-	    curl_setopt($curl_instance, CURLOPT_POST, 3);
-	    curl_setopt($curl_instance, CURLOPT_POSTFIELDS, $params);
-
-	    if ($cfg['sslverifyhost']) {
-		    curl_setopt($curl_instance, CURLOPT_SSL_VERIFYHOST, 2);
-	    } else {
-		    curl_setopt($curl_instance, CURLOPT_SSL_VERIFYHOST, 0);
-	    }
-	    if ($cfg['sslverifypeer']) {
-		    curl_setopt($curl_instance, CURLOPT_SSL_VERIFYPEER, 2);
-	    } else {
-		    curl_setopt($curl_instance, CURLOPT_SSL_VERIFYPEER, 0);
-	    }
-	    if(!$response = curl_exec($curl_instance)) {
-	        throw new SimpleSAML_Error_BadRequest("privacyIDEA: Bad request to PI server: " . curl_error($curl_instance));
-        };
-	    SimpleSAML_Logger::debug("privacyIDEA: \n\n\n" . $response . "\n\n\n");
-	    $header_size = curl_getinfo($curl_instance, CURLINFO_HEADER_SIZE);
-	    $body = json_decode(substr($response, $header_size));
-	    try {
-	        $result = $body->result;
-	        $status = $result->status;
-	        $value = $result->value->auth;
-        } catch (Exception $e) {
-	        throw new SimpleSAML_Error_BadRequest( "privacyIDEA: We were not able to read the response from the PI server");
+        if ($transaction_id) {
+            SimpleSAML_Logger::debug("Authenticating with transaction_id: " . $transaction_id);
+            $params["transaction_id"] = $transaction_id;
         }
-
-        if ($status !== True) {
-            throw new SimpleSAML_Error_BadRequest("Valid JSON Response, but some internal error occured in PI server");
-        } else {
-            if ($value !== True){
-                SimpleSAML_Logger::debug("privacyIDEA: Wrong user pass");
-                return False;
-            } else {
-	            SimpleSAML_Logger::debug("privacyIDEA: User authenticated successfully");
-                return True;
-            }
+        if ($signaturedata) {
+            SimpleSAML_Logger::debug("Authenticating with signaturedata: " . urlencode($signaturedata));
+            $params["signaturedata"] = $signaturedata;
+        }
+        if ($clientdata) {
+            SimpleSAML_Logger::debug("Authenticating with clientdata: " . urlencode($clientdata));
+            $params["clientdata"] = $clientdata;
         }
+	    $body = sspmod_privacyidea_Auth_utils::curl($params, null, $cfg, "/validate/samlcheck", true);
+		$attributes = NULL;
 
+	    try {
+		    $result = $body->result;
+		    $status = $result->status;
+		    $value  = $result->value;
+		    $auth   = $value->auth;
+	    } catch (Exception $e) {
+		    throw new SimpleSAML_Error_BadRequest("privacyIDEA: We were not able to read the response from the PI server");
+	    }
 
+	    if ($status !== true) {
+		    throw new SimpleSAML_Error_BadRequest("privacyIDEA: Valid JSON response, but some internal error occured in PI server");
+	    } else {
```

Comment:
Does ``throw`` in line 142 exit the code?

Then we can omit the else branch and have the ``if ($auth !== true)`` on the same level like ``if ($status !== true)``.

Reply:
fixed in https://github.com/privacyidea/simplesamlphp-module-privacyidea/pull/47/commits/1ceb254178130e64cb567121c15c7c3d0c46a781